### PR TITLE
Vb ab/model generating

### PR DESCRIPTION
**Pull request by Besermenji**
Code is ready for review, please be noted that because I did not add new migrations, but changed the current ones (because that was easiest for me and database was not formed completely), you will have to run <code>bundle exec rake db:purge db:migrate</code> to recreate the database.

UPDATE: 
Rebased it with master and now it can be merged automatically.

### CODE REVIEW

File: app/models/business_partner_account.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#  created_at          :datetime         not null
+#  updated_at          :datetime         not null
+#
+
+class BusinessPartnerAccount < ActiveRecord::Base
+  belongs_to :bank
+  belongs_to :business_partner
+
+  validates :number, presence: true,
+  					 uniqueness: true,
```
Comment (Besermenji) [inline app/models/business_partner_account.rb]: Will fix indentation here

File: app/models/company.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+#
+# Table name: companies
+#
+#  id              :integer          not null, primary key
+#  name            :string           not null
+#  contact_card_id :integer
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#
+
```
Comment (Besermenji) [inline app/models/company.rb]: And here

File: app/models/payment_proposal.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+#  created_at   :datetime         not null
+#  updated_at   :datetime         not null
+#
+
+class PaymentProposal < ActiveRecord::Base
+  has_many :payment_formings
+  has_many :payment_orders
+
+  validates :payment_date, presence: true
+  validates :number, presence: true,
```
Comment (Besermenji) [inline app/models/payment_proposal.rb]: And here

File: spec/factories/bank_factory.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+#
+#  id              :integer          not null, primary key
+#  name            :string           not null
+#  contact_card_id :integer
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#
+
+FactoryGirl.define do
+  factory :bank do
```
Comment (Besermenji) [inline spec/factories/bank_factory.rb]: And here

File: spec/factories/business_partner_account_factory.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+#  number              :string           not null
+#  activated           :boolean          default(FALSE)
+#  bank_id             :integer
+#  business_partner_id :integer
+#  created_at          :datetime         not null
+#  updated_at          :datetime         not null
+#
+
+FactoryGirl.define do
+  factory :business_partner_account do
```
Comment (Besermenji) [inline spec/factories/business_partner_account_factory.rb]: And Here

File: spec/factories/company_account_factory.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+#  number     :string           not null
+#  currency   :integer          not null
+#  company_id :integer
+#  bank_id    :integer
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+
+FactoryGirl.define do
+  factory :company_account do
```
Comment (Besermenji) [inline spec/factories/company_account_factory.rb]: And here

File: spec/factories/company_factory.rb [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+#
+#  id              :integer          not null, primary key
+#  name            :string           not null
+#  contact_card_id :integer
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#
+
+FactoryGirl.define do
+  factory :company do
```
Comment (Besermenji) [inline spec/factories/company_factory.rb]: Here too

File: spec/factories/contact_card_factory.rb [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+#  website    :string           not null
+#  phone      :string           not null
+#  fax        :string
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+
+FactoryGirl.define do
+  factory :contact_card do
+  	pib { Faker::Number.number(8) }
```
Comment (Besermenji) [inline spec/factories/contact_card_factory.rb]: Here..

File: spec/factories/daily_bank_statement_factory.rb [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+#  total_payment      :decimal(15, 4)
+#  total_payout       :decimal(15, 4)
+#  company_account_id :integer
+#  created_at         :datetime         not null
+#  updated_at         :datetime         not null
+#
+
+FactoryGirl.define do
+  factory :daily_bank_statement do
+  	number { Faker::Number.number(10) }
```
Comment (Besermenji) [inline spec/factories/daily_bank_statement_factory.rb]: Here

File: spec/factories/daily_statement_factory.rb [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+#  remaining_amount          :decimal(15, 4)   not null
+#  business_partner_id       :integer
+#  daily_bank_statement_id   :integer
+#  created_at                :datetime         not null
+#  updated_at                :datetime         not null
+#
+
+FactoryGirl.define do
+  factory :daily_statement do
+  	debtor_account_number { Faker::Number.number(18) }
```
Comment (Besermenji) [inline spec/factories/daily_statement_factory.rb]: Same here

File: spec/factories/financial_year_factory.rb [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+#  id         :integer          not null, primary key
+#  year       :string           not null
+#  closed     :boolean          default(FALSE)
+#  company_id :integer
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+
+FactoryGirl.define do
+  factory :financial_year do
```
Comment (Besermenji) [inline spec/factories/financial_year_factory.rb]: Here

File: config/environment.rb [hunk 12]
```diff
@@ -3,3 +3,121 @@
 
 # Initialize the Rails application.
 Rails.application.initialize!
+
+# currency constant
+CURRENCY_HASH = { 'ALL' => 'Albania Lek',
```
Comment (nikolalsvk) [inline config/environment.rb]: Is this really the place where we should store system constants? Just curious.

File: spec/factories/input_invoice_closure_factory.rb [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+#  closure_date       :datetime         not null
+#  closure_amount     :decimal(15, 4)   not null
+#  daily_statement_id :integer
+#  input_invoice_id   :integer
+#  created_at         :datetime         not null
+#  updated_at         :datetime         not null
+#
+
+FactoryGirl.define do
+  factory :input_invoice_closure do
```
Comment (Besermenji) [inline spec/factories/input_invoice_closure_factory.rb]: Here

File: spec/factories/input_invoice_factory.rb [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+#  circulation_date    :datetime         not null
+#  payment_deadline    :datetime         not null
+#  business_partner_id :integer
+#  financial_year_id   :integer
+#  created_at          :datetime         not null
+#  updated_at          :datetime         not null
+#
+
+FactoryGirl.define do
+  factory :input_invoice do
```
Comment (Besermenji) [inline spec/factories/input_invoice_factory.rb]: Here

File: spec/factories/output_invoice_closure_factory.rb [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+#  closure_date       :datetime         not null
+#  closure_amount     :decimal(15, 4)   not null
+#  daily_statement_id :integer
+#  output_invoice_id  :integer
+#  created_at         :datetime         not null
+#  updated_at         :datetime         not null
+#
+
+FactoryGirl.define do
+  factory :output_invoice_closure do
```
Comment (Besermenji) [inline spec/factories/output_invoice_closure_factory.rb]: Here

File: spec/factories/output_invoice_factory.rb [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+#  payment_deadline    :datetime         not null
+#  company_id          :integer
+#  business_partner_id :integer
+#  financial_year_id   :integer
+#  created_at          :datetime         not null
+#  updated_at          :datetime         not null
+#
+
+FactoryGirl.define do
+  factory :output_invoice do
```
Comment (Besermenji) [inline spec/factories/output_invoice_factory.rb]: Here

File: spec/factories/payment_forming_factory.rb [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+#  amount              :decimal(15, 4)   not null
+#  company_account_id  :integer
+#  input_invoice_id    :integer
+#  payment_proposal_id :integer
+#  created_at          :datetime         not null
+#  updated_at          :datetime         not null
+#
+
+FactoryGirl.define do
+  factory :payment_forming do
```
Comment (Besermenji) [inline spec/factories/payment_forming_factory.rb]: Here

File: spec/factories/payment_order_factory.rb [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+#  currency_date             :datetime         not null
+#  payment_date              :datetime         not null
+#  account_city              :string           not null
+#  payment_forming_id        :integer
+#  created_at                :datetime         not null
+#  updated_at                :datetime         not null
+#
+
+FactoryGirl.define do
+  factory :payment_order do
```
Comment (Besermenji) [inline spec/factories/payment_order_factory.rb]: Here

File: spec/rails_helper.rb [hunk 19]
```diff
@@ -55,3 +55,22 @@
...
+
+    # Choose one or more libraries:
+    with.library :active_record
+    with.library :active_model
+    with.library :action_controller
+    # Or, choose the following (which implies all of the above):
+    with.library :rails
+  end
+end
+
```
Comment (nikolalsvk) [inline spec/rails_helper.rb]: Just remove these extra lines here.

**Comment by nikolalsvk** [comment]
I'm trusting you @Besermenji @aleksandarbos that this is a valid representation of our model. We can rework it later if needed. Good job :clap: :clap: 

Also, you'll have to rebase it with `master` branch, I can help you out with that.

**Pull request merged**